README.md

```diff
@@ -163,19 +215,32 @@ produce early errors because of recursive application of the first rule.
         :: MemberExpression
 
 - Let _targetReference_ be the result of evaluating _MemberExpression_.
-- Assert: Type(_targetReference_) is Reference.
+- Assert: Type(_targetReference_) is `Reference`.
 - Let _thisValue_ be GetThisValue(_targetReference_).
 - Let _target_ be GetValue(_targetReference_).
 - If IsCallable(_target_) is `false`, throw a `TypeError` exception.
-- Let _F_ be ? BoundFunctionCreate(_target_, _thisValue_, «»).
-- Let _targetHasLength_ be ? HasOwnProperty(_target_, `"length"`).
-- If _targetHasLength_ is **true**, then
-    - Let _targetLen_ be ? Get(_target_, `"length"`).
-    - If Type(_targetLen_) is not `Number`, then let _L_ be `0`.
-    - Else, let _L_ be ToInteger(_targetLen_).
-- Else let _L_ be `0`.
-- Let _status_ be ? DefinePropertyOrThrow(_F_, `"length"`, PropertyDescriptor {[[Value]]: _L_, [[Writable]]: `false`, [[Enumerable]]: `false`, [[Configurable]]: `true`}).
-- Let _targetName_ be ? Get(_target_, `"name"`).
-- If Type(_targetName_) is not `String`, then let _targetName_ be the empty string.
-- Let _status_ be ? SetFunctionName(_F_, _targetName_, `"bound"`).
+- Let _F_ be BoundFunctionCreate(_target_, _thisValue_, «»).
+- Perform InitBoundProperties(_F_, _target_).
+- Return _F_.
+
+
+----
+
+    BindExpression:
+        LeftHandSideExpression[?Yield] :: new
+
+- Let _targetReference_ be the result of evaluating _LeftHandSideExpression[?Yield]_.
+- Assert: Type(_targetReference_) is `Reference`.
+- Assert: IsSuperReference(_targetReference_) is `false`.
+- Let _target_ be GetValue(_targetReference_).
+- If IsConstructible(_target_) is `false`, throw a `TypeError` exception.
+- Let _bound_ be a new bound constructor wrapper function as defined in Bound Constructor Wrapper Functions.
+- Set the [[BoundTargetConstructor]] internal slot of _bound_ to _target_.
+- Let _desc_ be _target_.\[\[GetOwnProperty\]\]("caller")
+- ReturnIfAbrupt(_desc_).
+- If _desc_ is not `undefined`, then
+    - Let _status_ be DefinePropertyOrThrow(_bound_, "caller", _desc_).
+    - Assert: _status_ is not an abrupt completion.
+- Let _F_ be BoundFunctionCreate(_bound_, `undefined`, «»).
```

Comment:
I'm having trouble parsing what you said. Do you mean returning a function wrapper like what I did for constructors, or something else?

Also, BoundFunctionCreate is half-irrelevant, since the common case is just sugar for `Function.prototype.call`, and the function binding case is almost this in practice:

```js
function bind(f, obj) {
  if (!Reflect.isCallable(C)) {
    throw new TypeError("Expected a function")
  }
  return function () {
    return f.apply(obj, arguments)
  }
}
```

The constructor case is almost literally this:

```js
function bind(C) {
  if (!Reflect.isConstructor(C)) {
    throw new TypeError("Expected a constructor")
  }
  return function (...args) {
    return new C(...args)
  }
}
```